### PR TITLE
Move EventRegistrationToken to shared

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/coreclr/issues/17903

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices.WindowsRuntime
+{
+    // Event registration tokens are 64 bit opaque structures returned from WinRT style event adders, in order
+    // to signify a registration of a particular delegate to an event.  The token's only real use is to
+    // unregister the same delgate from the event at a later time.
+    public struct EventRegistrationToken : IEquatable<EventRegistrationToken>
```
Comment (marek-safar) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: Could the struct be changed to be readonly?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        internal readonly ulong _value;
+
+        internal EventRegistrationToken(ulong value) => _value = value;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: ProjectN needs the `ctor` & `Value` to be public. They are called from `System.Private.Interop`.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices.WindowsRuntime
+{
+    // Event registration tokens are 64 bit opaque structures returned from WinRT style event adders, in order
+    // to signify a registration of a particular delegate to an event.  The token's only real use is to
+    // unregister the same delgate from the event at a later time.
+    public struct EventRegistrationToken : IEquatable<EventRegistrationToken>
+    {
+        internal readonly ulong _value;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: Can this be `private` ?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: Yes, but I don't want to change public surface area as part of this.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: Ok

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs (hunk 3 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/WindowsRuntime/EventRegistrationToken.cs]: Don't see why not.

**Pull request merged**